From bd987536cb3b56ba51831b2ce51e7856e589e695 Mon Sep 17 00:00:00 2001
From: Patryk Obara <dreamer.tan@gmail.com>
Date: Sun, 7 Mar 2021 00:44:57 +0100
Subject: [PATCH 1/2] Remove "-git" suffix from .conf file in release

Default .conf file name after this change is:

- dosbox-staging.conf - for buildtypes release, plain, and other
  non-debug build types
- dosbox-staging-git.conf - for buildtype debug and debugoptimized

It does not work for Visual Studio buildsystem, but we're distributing
these builds ourselves, so simple adjustment on CI is enough (this tiny
hack will disappear once VS buildsystem will be replaced with Meson
anyway).

Pros of implementing this change:

- It's easier to test Linux and macOS builds before the release
- Removes another step from our release checklist (!)
- Makes life easier for packagers, who want to deliver pre-release
  builds (e.g. preparing new package or providing weekly test builds)

Cons of implementing this change:

- Users testing our official snapshot builds might end up with broken
  default .conf file for next stable release.
---
 .github/workflows/windows.yml | 5 ++++-
 meson.build                   | 6 ++++++
 src/config.h.in               | 4 ++--
 3 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/.github/workflows/windows.yml b/.github/workflows/windows.yml
index 9d541e275..0a9eb0c2c 100644
--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -108,14 +108,17 @@ jobs:
         shell: pwsh
         run:   .\scripts\log-env.ps1
 
-      - name:  Inject version string
+      - name:  Adjust config.h
         shell: bash
         run: |
           set -x
           git fetch --prune --unshallow
           export VERSION=$(git describe --abbrev=4)
+          # inject version based on vcs
           sed -i "s|DOSBOX_DETAILED_VERSION \"git\"|DOSBOX_DETAILED_VERSION \"$VERSION\"|" src/platform/visualc/config.h
           echo "VERSION=$VERSION" >> $GITHUB_ENV
+          # overwrite .conf file branding for release build
+          sed -i "s|CONF_BRAND \"staging-git\"|CONF_BRAND \"staging\"|" src/platform/visualc/config.h
 
       - name:  Build
         shell: pwsh
diff --git a/meson.build b/meson.build
index 745a7d636..5a76584bc 100644
--- a/meson.build
+++ b/meson.build
@@ -35,6 +35,12 @@ endif
 conf_data = configuration_data()
 conf_data.set('version', meson.project_version())
 
+if get_option('buildtype').startswith('debug')
+  conf_data.set('conf_branding', 'staging-git')
+else
+  conf_data.set('conf_branding', 'staging')
+endif
+
 os_family_name = {
   'linux'     : 'LINUX',
   'windows'   : 'WIN32',
diff --git a/src/config.h.in b/src/config.h.in
index 0e9c04b58..c0ea73ccd 100644
--- a/src/config.h.in
+++ b/src/config.h.in
@@ -27,8 +27,8 @@
 // Emulator version
 #define VERSION "@version@"
 
-// TODO make it a suffix, configurable via meson
-#define CONF_BRAND "staging-git"
+// Suffix appended to the default .conf file name
+#define CONF_BRAND "@conf_branding@"
 
 /* Operating System
  */

From 7ab60c7f7bdcb3d261900317ddd51c50a1d3a1a4 Mon Sep 17 00:00:00 2001
From: Patryk Obara <dreamer.tan@gmail.com>
Date: Wed, 10 Mar 2021 11:45:12 +0100
Subject: [PATCH 2/2] Rename CONF_BRAND to CONF_SUFFIX

---
 .github/workflows/windows.yml | 2 +-
 meson.build                   | 4 ++--
 src/config.h.in               | 2 +-
 src/misc/cross.cpp            | 2 +-
 src/platform/visualc/config.h | 2 +-
 5 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/.github/workflows/windows.yml b/.github/workflows/windows.yml
index 0a9eb0c2c..ccc520c1d 100644
--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -118,7 +118,7 @@ jobs:
           sed -i "s|DOSBOX_DETAILED_VERSION \"git\"|DOSBOX_DETAILED_VERSION \"$VERSION\"|" src/platform/visualc/config.h
           echo "VERSION=$VERSION" >> $GITHUB_ENV
           # overwrite .conf file branding for release build
-          sed -i "s|CONF_BRAND \"staging-git\"|CONF_BRAND \"staging\"|" src/platform/visualc/config.h
+          sed -i "s|CONF_SUFFIX \"-staging-git\"|CONF_SUFFIX \"-staging\"|" src/platform/visualc/config.h
 
       - name:  Build
         shell: pwsh
diff --git a/meson.build b/meson.build
index 5a76584bc..1e8d287d7 100644
--- a/meson.build
+++ b/meson.build
@@ -36,9 +36,9 @@ conf_data = configuration_data()
 conf_data.set('version', meson.project_version())
 
 if get_option('buildtype').startswith('debug')
-  conf_data.set('conf_branding', 'staging-git')
+  conf_data.set('conf_suffix', '-staging-git')
 else
-  conf_data.set('conf_branding', 'staging')
+  conf_data.set('conf_suffix', '-staging')
 endif
 
 os_family_name = {
diff --git a/src/config.h.in b/src/config.h.in
index c0ea73ccd..13a3bfa57 100644
--- a/src/config.h.in
+++ b/src/config.h.in
@@ -28,7 +28,7 @@
 #define VERSION "@version@"
 
 // Suffix appended to the default .conf file name
-#define CONF_BRAND "@conf_branding@"
+#define CONF_SUFFIX "@conf_suffix@"
 
 /* Operating System
  */
diff --git a/src/misc/cross.cpp b/src/misc/cross.cpp
index 2121fb18e..211fe69c9 100644
--- a/src/misc/cross.cpp
+++ b/src/misc/cross.cpp
@@ -46,7 +46,7 @@
 
 static std::string GetConfigName()
 {
-	return "dosbox-" CONF_BRAND ".conf";
+	return "dosbox" CONF_SUFFIX ".conf";
 }
 
 #ifndef WIN32
diff --git a/src/platform/visualc/config.h b/src/platform/visualc/config.h
index 4b1713eb6..885b535e3 100644
--- a/src/platform/visualc/config.h
+++ b/src/platform/visualc/config.h
@@ -1,5 +1,5 @@
 /* String appended to the .conf file. */
-#define CONF_BRAND "staging-git"
+#define CONF_SUFFIX "-staging-git"
 
 /* Version number of package */
 #define VERSION "0.77.0"
